NGitLab.Mock/Repository.cs

```diff
@@ -204,12 +204,12 @@ public Commit Commit(CommitCreate commitCreate)
         var mustCreateBranch = !repo.Commits.Any();
         if (!mustCreateBranch)
         {
-            Commands.Checkout(repo, commitCreate.Branch);
+            Commands.Checkout(repo, commitCreate.StartBranch ?? commitCreate.StartSha);
```

Comment:
There are problems with this change. But to be fair, there were already issues with the existing mock code...

I just added a few more commit creation tests in the main branch, to document GitLab's behavior. We should try to reproduce that behavior in the mocks as much as possible. @jasonwoods-7, could you rebase your branch, look up the aforementioned tests and adapt your PR accordingly? Similar tests should be added for the mocks.

Reply:
ok, i think i have all the cases covered now. let me know if i missed any.